parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -524,18 +507,24 @@ def start(self):
         thr_monitoring_handler = threading.Thread(
             target=self.handle_monitoring_messages, name="Monitoring-Handler"
         )
+        thr_heartbeater = threading.Thread(target=self.heartbeater, name="Heartbeater")
 
         thr_task_puller.start()
         thr_result_pusher.start()
         thr_worker_watchdog.start()
         thr_monitoring_handler.start()
+        thr_heartbeater.start()
 
         logger.info("Manager threads started")
 
         # This might need a multiprocessing event to signal back.
         self._stop_event.wait()
         logger.critical("Received kill event, terminating worker processes")
 
+        # Invite blocking threads to quit
+        self.pending_result_queue.put(None)
+
+        thr_heartbeater.join()
```

Comment:
i haven't verified that this exit really happens - in real life worker pools are usually killed by external termination and I'm not sure how well tested this other path is before this PR.

Reply:
Meaning, add a test?  I can do that.  However, for that precise reason, I also wasn't immediately worried about this exit dance, especially in light of the flux that is currently happening in this file.  The larger point of interest for me was the unnecessary iteration during idle.  If there's nothing to do ... then don't do anything.

Reply:
meaning i don't particularly care but was noting that i don't care

Reply:
So ... we're of similar mind.  You just chose to make that known.  :stuck_out_tongue: